codefresh.yml

```diff
@@ -0,0 +1,41 @@
+version: '1.0'
+
+steps:
+
+  agent_test:
+    image: node:7.2.0
+    working_directory: ${{main_clone}}/agent
+    commands:
+      - npm i -g yarn
+      - yarn install
+      - yarn test
+
+  agent_build:
+    type: build
+    working_directory: ${{main_clone}}/agent
+    tag: latest
+    image_name: httpmark-test-agent
+    when:
+      branch:
+        only: [ master ]
+
+  agent_push:
+    type: push
+    candidate: ${{agent_build}}
+    provider: ecr
+    tag: latest
+    accessKeyId: ${{AWS_ACCESS_KEY_ID}}
+    secretAccessKey: ${{AWS_SECRET_ACCESS_KEY}}
+    region: eu-west-1
+    when:
+      branch:
+        only: [ master ]
+
+  app_test:
+    image: node:7.2.0
+    working_directory: ${{main_clone}}/app
+    commands:
+      - npm i -g yarn
+      - npm i -g elm@0.18.0
+      - cd ui && yarn install
```

Comment:
Actually, my bad, it doesn't, but it should - the makefile should be written in a way to trigger a build before test if necessary.

It's fairly easy to write the build step in terms of the build output (the bundles) and then make test dependent on that. Can we change? (In my mind the simpler the CI script the better and this sort of logic should be handled by the makefiles and npm scripts).